### PR TITLE
ENH Matt's Edits

### PR DESCRIPTION
**Pull request by beckermr**
This PR has my edits.

### CODE REVIEW

File: posts/2020-07-11-R-4.rst [hunk 1]
```diff
@@ -10,11 +9,26 @@
...
+its dependents gives a good opportunity to write a retrospective on the technical issues with large-scale migrations 
+in ``conda-forge`` and how we solved them. 
 
-The enhancements that made this possible were spearheaded by Matt Becker and include:
+The R 4.0 migration rebuilt every package in ``conda-forge`` that had ``r-base`` as a requirement, including more 
+than 2200 feedstocks. A migration of this size in ``conda-forge`` faces several hurdles. First, since every feedstock
+is a separate GitHub repository, one needs to merge more 2200 pull requests (PRs). Second, ``conda-forge``'s packages
+on ``anaconda.org`` are behind a CDN (content delivery network). This service reduces web hosting costs for Anaconda Inc.
+but introduces an approximately 30 minute delay from when a package is uploaded to ``anaconda.org`` and when it will 
+appear as available using ``conda`` from the command line. Thus even if the dependencies of a package have been built,
```
Comment (CJ-Wright) [inline posts/2020-07-11-R-4.rst]:
```suggestion
appear as available using ``conda`` from the command line. Thus, even if the dependencies of a package have been built, 
```

File: posts/2020-07-11-R-4.rst [hunk 2]
```diff
@@ -10,11 +9,26 @@
...
+but introduces an approximately 30 minute delay from when a package is uploaded to ``anaconda.org`` and when it will 
+appear as available using ``conda`` from the command line. Thus even if the dependencies of a package have been built, 
+we have to wait until they appear on the CDN before we can successfully issue the next PR and have it build correctly. 
+Finally, the existing bot and ``conda`` infrastructure limited the throughput of the migrations, due in part to the speed 
+of the ``conda`` solver.
+
+Given the size of the R 4.0 migration, we took this opportunity to try out a bunch of new technology to speed up large-scale 
+migrations. The main enhancements were using GitHub Actions to automerge PRs, using ``mamba`` to quickly check for solvability 
+of package environments, and enabling long-running migration jobs for the autotick bot. All told, the bulk of the feedstocks 
+for R 4.0 were rebuilt in less than a week, with many PRs being merged in 30 minutes or less from when they were issued. These
```
Comment (CJ-Wright) [inline posts/2020-07-11-R-4.rst]: we have plots of this if we want to use them.

**Comment by beckermr** [comment]
OK @CJ-Wright. This one is ready for review and merging once we are happy with it.

File: posts/2020-07-11-R-4.rst [hunk 3]
```diff
@@ -9,65 +9,86 @@
...
-This approach will help to provide the latest features to downstream users and keep transition times to a minimum, helping to foster greater stability of the ecosystem and the seamless experience users have come to expect from conda-forge.
+Automerging conda-forge PRs
+---------------------------
+
+In a typical migration on ``conda-forge``, we issue a PR to a feedstock and then ask the feedstock maintainers to
+make sure it passes and merge it. In the case of the R 4.0 migration, the maintainers of R packages on ``conda-forge``
+use a maintenance team (i.e., ``@conda-forge/r``) on the vast majority of feedstocks. This team is small and so merging
+over 2000 PRs by hand is a big undertaking. Thus, with their permission, we added the ``conda-forge`` automerge
+functionality to all R feedstocks that they maintain. The automerge bot, which relies on GitHub Actions, is able to
+automatically merge any PR form the autotick bot that passes the recipe linter, the continuous integration services,
```
Comment (CJ-Wright) [inline posts/2020-07-11-R-4.rst]:
```suggestion
automatically merge any PR from the autotick bot that passes the recipe linter, the continuous integration services,
```

File: posts/2020-07-11-R-4.rst [hunk 4]
```diff
@@ -9,65 +9,86 @@
...
+Despite some initial hiccups with the bot infrastructure, the migration ran quite smoothly for an endeavor of
+its size. The vast majority of migration PRs were completed within a week from when we started, which is a first
+for a migration of this size on ``conda-forge``. The largest issue was solved recently, with the fixing of the
+``openjdk`` recipe and the removal of ``aarch64`` and ``ppc64le`` builds from ``r-java``, enabling the last
+large piece of the R ecosystem to be updated.
+
+Looking forward, the improvements we made for the R 4.0 migration seem broadly applicable to other migration tasks,
+including the yearly python minor version bump. These kinds of large-scale migrations are particularly suitable,
+since they usually involve few changes to the feedstock itself and usually fail on CI when a broken package
+would be produced. Faster migrations approach will help to provide the latest features to downstream users and
```
Comment (CJ-Wright) [inline posts/2020-07-11-R-4.rst]:
```suggestion
would be produced. Faster migrations will help to provide the latest features to downstream users and
```

**Comment by CJ-Wright** [comment]
@beckermr this looks great! Some minor copyedit suggestions then I think this can go in!

**Comment by beckermr** [comment]
done! ready to merge!

**Pull request merged**